format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java

```diff
@@ -243,7 +284,14 @@ protected ValidatingInputFormat getInputFormatForRun(BatchSourceContext context)
           + "or re-create the pipeline with all required properties.",
         fileFormat, properties);
       throw new IllegalArgumentException(errorMessage, e);
+    } catch (InstantiationException e) {
+      context.getFailureCollector().addFailure(
```

Comment:
create a local variable `collector = context.getFailureCollector()` and use at other places in same method.